File: Object-C/BlackFireIM/Example/Lib/MSIMSDK-UIKit/Location/MSLocationListCell.m
//
//  MSLocationListCell.m
//  BlackFireIM
//
//  Created by benny wang on 2021/11/29.
//

#import "MSLocationListCell.h"
#import "MSIMSDK-UIKit.h"


@interface MSLocationListCell()



@end
@implementation MSLocationListCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        self.backgroundColor = [UIColor d_colorWithColorLight:TCell_Nomal dark:TCell_Nomal_Dark];
        [self.contentView addSubview:self.titleL];
        [self.contentView addSubview:self.addressL];
        [self.contentView addSubview:self.checkIcon];
        
    }
    return self;
}

- (void)configCell:(MSLocationInfo *)info
{
    self.titleL.text = info.name;
    self.addressL.text = [NSString stringWithFormat:@"%zd m | %@%@",info.distance,XMNoNilString(info.district),XMNoNilString(info.address)];
    self.checkIcon.hidden = !info.isSelect;
}

- (UILabel *)titleL
{
    if (!_titleL) {
        _titleL = [[UILabel alloc]init];
        _titleL.font = [UIFont systemFontOfSize:16];
        _titleL.textColor = [UIColor d_colorWithColorLight:TText_Color dark:TText_Color_Dark];
    }
    return _titleL;
}

- (UILabel *)addressL
{
    if (!_addressL) {
        _addressL = [[UILabel alloc]init];
        _addressL.font = [UIFont systemFontOfSize:13];
        _addressL.textColor = [UIColor d_systemGrayColor];
    }
    return _addressL;
}

- (UIImageView *)checkIcon
{
    if (!_checkIcon) {
        _checkIcon = [[UIImageView alloc]init];
        _checkIcon.image = [UIImage imageNamed:TUIKitResource(@"ic_selected")];
    }
    return _checkIcon;
}

- (void)layoutSubviews
{
    [super layoutSubviews];
    self.titleL.frame = CGRectMake(15, 15, Screen_Width - 30 - 40, 16);
    self.addressL.frame = CGRectMake(15, self.titleL.maxY + 10, self.titleL.width, 13);
    self.checkIcon.frame = CGRectMake(Screen_Width - 15 - 22, self.height * 0.5 - 11, 22, 22);
}

@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/Location/MSLocationListCell.swift
//
//  MSLocationListCell.swift
//  MSIM-Swift
//
//  Created by benny wang on 2021/12/9.
//

import UIKit

class MSLocationListCell: UITableViewCell {

    public var locationInfo:  MSLocationInfo! {
        didSet {
            titleL.text = locationInfo.name
            addressL.text = "\(locationInfo.distance) m | \(locationInfo.district)\(locationInfo.address)"
            checkIcon.isHidden = !locationInfo.isSelect
        }
    }
    
    lazy private var titleL: UILabel = {
        let titleL = UILabel()
        titleL.font = .systemFont(ofSize: 16)
        titleL.textColor = UIColor.d_color(light: MSMcros.TText_Color, dark: MSMcros.TText_Color_Dark)
        return titleL
    }()
    
    lazy private var addressL: UILabel = {
        let addressL = UILabel()
        addressL.font = .systemFont(ofSize: 13)
        addressL.textColor = .systemGray
        return addressL
    }()
    
    lazy private var checkIcon: UIImageView = {
        let checkIcon = UIImageView()
        checkIcon.image = UIImage.bf_imageNamed(name: "ic_selected")
        return checkIcon
    }()
    
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.backgroundColor = UIColor.d_color(light: MSMcros.TCell_Nomal, dark: MSMcros.TCell_Nomal_Dark)
        
        contentView.addSubview(titleL)
        contentView.addSubview(addressL)
        contentView.addSubview(checkIcon)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        titleL.frame = CGRect(x: 15, y: 15, width: UIScreen.width - 30 - 40, height: 16)
        addressL.frame = CGRect(x: 15, y: titleL.bottom + 10, width: titleL.width, height: 13)
        checkIcon.frame = CGRect(x: UIScreen.width - 15 - 22, y: self.height * 0.5 - 11, width: 22, height: 22)
    }
}
